noir-projects/noir-contracts/contracts/token_contract/src/main.nr

```diff
@@ -335,13 +336,38 @@ contract Token {
         let to_ivpk = header.get_ivpk_m(&mut context, to);
 
         let amount = U128::from_integer(amount);
-        storage.balances.sub(from, amount).emit(encode_and_encrypt_note_with_keys_unconstrained(&mut context, from_ovpk, from_ivpk));
+        let mut (change, missing) = storage.balances.odd_sub(from, amount, 2);
+
+        // If we are still missing some balance to cover the amount, call accumulate to spend more notes.
+        let call = Token::at(context.this_address())._accumulate(from, missing.to_field());
+        if !missing.eq(U128::zero()) {
+            change = call.call(&mut context);
+        }
+
+        storage.balances.add(from, change).emit(encode_and_encrypt_note_with_keys_unconstrained(&mut context, from_ovpk, from_ivpk));
```

Comment:
```suggestion
        // We don't constrain encryption of the note log in `transfer` (unlike in `transfer_from`) because transfer
        // function is only designed to be used in situations where it's not necessary (e.g. payment to another person
        // where the payment is considered to be successful when the other party successfully decrypts a note).
        storage.balances.add(from, change).emit(encode_and_encrypt_note_with_keys_unconstrained(&mut context, from_ovpk, from_ivpk));
```